File: app/src/main/java/it/unical/mat/embasp/base/OptionDescriptor.java
package it.unical.mat.embasp.base;

/** Represents options for a generic ASP programs */
public class OptionDescriptor {
	/** used as option separator */
	protected String separator;
	/** where options are stored */
	protected String options;

	public OptionDescriptor() {
		options = new String();
		separator = new String();
	}

	public OptionDescriptor(final String initial_option) {
		this();
		options = new String(initial_option);
	}

	/**
	 * concatenate a new option in a String format to the current {@link #options}
	 *
	 * @param option
	 *            String to be concatenated
	 */
	public void addOption(final String option) {
		if (option.isEmpty())
			options = option;
		else
			options += separator + option;

	}

	/** after using this method the {@link #options} variable will be empty */
	public void clear() {

		options = "";
	}

	/**
	 * Returns values stored in {@link #options}
	 *
	 * @return {@link #options}'s data in a String format
	 */
	public String getOptions() {

		return options;
	}

	/** get separator character */
	public String getSeparator() {
		return separator;
	}

	public void setOptions(final String option) {
		options = option;
	}

	/** Set {@link #separator} character with new separator */
	public void setSeparator(final String separator) {
		this.separator = separator;
	}
}

File: Python version/base/option_descriptor.py
class OptionDescriptor(object):
    """Represents options for a generic ASP programs."""

    def __init__(self, initial_option=None):
        self._options = initial_option  # Where options are stored
        self._separator = ""  # Used as option separator

    def add_option(self, option):
        """Concatenate a new option in a string format to the current _options.

        The parameter option is the string to be concatenated.
        """
        if self._options is None or self._options == "":
            self.set_options(option)
        else:
            self._options += self._separator + option

    def clear(self):
        """After using this method the _options variable will be empty."""
        self._options = ""

    def get_options(self):
        """Returns values stored in _options, in a string format."""
        return self._options

    def get_separator(self):
        """Get separator character."""
        return self._separator

    def set_options(self, option):
        """Set _option string with new string."""
        self._options = option

    def set_separator(self, separator):
        """Set _separator character with new separator."""
        self._separator = separator
